pkg/compactor/block_upload.go

```diff
@@ -365,6 +401,75 @@ func (c *MultitenantCompactor) uploadMeta(ctx context.Context, logger log.Logger
 	return nil
 }
 
+func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.ULID,
+	userBkt objstore.Bucket, meta metadata.Meta) error {
+	blockDir, err := os.MkdirTemp(filepath.Join(c.compactorCfg.DataDir), "upload")
+	if err != nil {
+		return errors.Wrap(err, "failed to create temp dir")
+	}
+	defer func() {
+		if err := os.RemoveAll(blockDir); err != nil {
+			level.Warn(c.logger).Log("msg", "failed to remove temp dir", "path", blockDir, "err", err)
+		}
+	}()
+
+	var objectCount, objectBytes int64
+	if err := userBkt.Iter(ctx, blockID.String(), func(pth string) error {
+		fname := filepath.Join(blockDir, pth)
+		if pth == uploadingMetaFilename {
+			fname = filepath.Join(blockDir, block.MetaFilename) // rename to final meta.json in temp dir
+		}
+
+		r, err := userBkt.Get(ctx, pth)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("failed to get object %q from object storage", pth))
+		}
+
+		f, err := os.Create(fname)
+		if err != nil {
+			return errors.Wrap(err, "failed creating temp file")
+		}
+		defer func() {
+			_ = f.Close()
+		}()
+		bytesWritten, err := io.Copy(f, r)
+		if err != nil {
+			return err
+		}
+		objectCount++
+		objectBytes += bytesWritten
+		if err := f.Close(); err != nil {
+			return errors.Wrap(err, "failed to close temp file")
+		}
+		return nil
+	}); err != nil {
+		return errors.Wrapf(err, "failed to iterate block %s", blockID.String())
+	}
+
+	// Read metadata file, populate mop of file paths and sizes
```

Comment:
```suggestion
	// Read metadata file, populate map of file paths and sizes
```

Reply:
I realized I don't need this comment anymore, it refers to deleted code.